client/src/containers/Gene/BaseForm.js

```diff
@@ -84,14 +84,15 @@ class FormDataStore {
       (result, fieldId) => {
         const value = fields[fieldId] && fields[fieldId].value;
         const idSegments = fieldId.split(':');
-        return {
-          ...idSegments.reduceRight((result, idSegment, index) => {
-            return {
-              [idSegment]: result,
-            };
-          }, value),
-          ...result,
-        };
+        idSegments.reduce((resultSubtree, idSegment, index) => {
+          if (index < idSegments.length - 1) {
+            resultSubtree[idSegment] = resultSubtree[idSegment] || {};
```

Comment:
maybe `resultSubtree[idSegment] |=  {}`

Reply:
I think javascript may not have the `||=` or `|=` syntax. I get syntax error when I tried that.

Reply:
hur seemed like it did to me, but is a minor thing anyway (https://www.w3schools.com/js/js_assignment.asp)

Reply:
hmm, yeah, only seems to work for numbers (logical bit-wise assignment)